guvnor-structure/guvnor-structure-client/src/test/java/org/guvnor/structure/client/editors/repository/clone/mock/MockCloneRepositoryPresenter.java

```diff
@@ -0,0 +1,18 @@
+package org.guvnor.structure.client.editors.repository.clone.mock;
+
+import org.guvnor.structure.client.editors.repository.clone.CloneRepositoryPresenter;
+
+public class MockCloneRepositoryPresenter extends CloneRepositoryPresenter {
+
+    private boolean isOuMandatory;
+
+    public MockCloneRepositoryPresenter(boolean isOuMandatory) {
```

Comment:
If you change to use constructor injection in the Presenter you will not need to provide a "mock" method overriding the ```isOUMandatory()``` method as you can pass a mock ```RepositoryPreferences``` object to the constructor and set it up to return true/false depending on the test.